service/src/main/java/bio/terra/landingzone/library/landingzones/management/ResourcesReader.java

```diff
@@ -11,5 +11,9 @@ public interface ResourcesReader {
 
   List<DeployedResource> listResourcesByPurpose(ResourcePurpose purpose);
 
+  List<DeployedResource> listResourcesWithPurpose();
+
   List<DeployedVNet> listVNetWithSubnetPurpose(SubnetResourcePurpose purpose);
+
+  List<DeployedVNet> listVNetResourcesWithPurpose();
```

Comment:
I don’t think this method is necessary. In the service you can enumerate the all the purposes and call listResourceByPurpose for each and construct the desired result from there.

Reply:
That is basically exactly what is does: reading all the VNet with subnet/purposes existing for all subnet purposes. I prefer encapsulating the logic of reading resources in the reader implementation.